articles/kotlin-ranges/index.md

```diff
@@ -0,0 +1,134 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/kotlin-ranges/
+title: Introduction to Kotlin Ranges
+description: This article we will look at Kotlin ranges, and use functions to create progressions and check for certain values in those ranges.
+author: linus-muema
+date: 2020-10-21T00:00:00-14:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/kotlin-ranges/hero.png
+    alt:  image example
+---
+Kotlin as a programming language provides a wide range of solutions. Most of these solutions aim to make development easier and faster. These solutions provide easier ways to perform specific actions when using Kotlin.
+<!--more-->
+### Kotlin Range Operators
+Since Kotlin is compiled the same way as [Java](https://en.wikipedia.org/wiki/Java_(programming_language)), its functions and methods ensure your code is efficient. The boilerplate code present in Java is also greatly reduced. This article addresses one of the solutions it offers: iterations.
+
+In this article, we will take a look at Kotlin ranges.
+
+### What are ranges
+A range is a collection of values determined by a start and stop value. A simple example is, values between 1 and 5 form a range. That is, a range of 1 as the start value and 5 as the end value. Both the start and end values belong to the range as well. In other words, a range is an interval from the start value to end value. We can use functions to create progressions and check for certain values in those ranges. Let's look at how we could apply ranges.  
+
+Before we begin writing code, you can use the following [repl.it](https://repl.it/@Linusmuema/Ranges) to run the functions.
+
+All you need to do is call the specific function in the `main` function. Then run the code and your output will be printed 🤓.
+
+For instance, if we want to run the `printNumRange` function:
+
+```Kotlin
+  fun main(args: Array<String>) {
+      // call the function here
+      printNumRange()
+  }
+```
+
+### 1. To Create a Progression
+We can use the range operator to create an iterable range. A loop is created that goes through each item from the start value to the end value. This helps us to be sure of the number of iterations in the loop. It gives the developer more control over the loop. In the repl.it, call the `printNumRange` function and run it. The output should be:
+
```

Comment:
Please include the `printNumRange` function here.

Reply:
added the function definition